evm/src/executor/inspector/cheatcodes/mod.rs

```diff
@@ -124,6 +128,17 @@ pub struct Cheatcodes {
     /// Expected revert information
     pub expected_revert: Option<ExpectedRevert>,
 
+    /// Stores the expected reverts associated to an address
+    pub expected_reverts_with_address: VecDeque<ExpectedRevertWithAddress>,
+
+    /// Tracks if all the expected reverts associated to an address have been matched to alter the
+    /// root call return data from a revert to a simple return
+    pub matched_all_expected_reverts_with_address: bool,
```

Comment:
If we need a `vecdequeue` here, can't we just map over it any see if all have been matched with an `.any()`?

Reply:
I used that boolean because I want to explicitly distinguish the scenario where the queue was empty from the start from the one where all the reverts have been matched and the queue was cleared. I can't simply check if the deque is empty because if the user did not use `expectRevert` with an address, modifying the return value here would be wrong.

![image](https://github.com/foundry-rs/foundry/assets/42520800/afeaeef5-fce3-405c-934a-ab04cb597b5f)